src/lib/navigation/navigation-mode.interface.ts

```diff
@@ -1,64 +1,81 @@
+   *
    * @param destinationIndex The index of the destination step
    * @returns A [[Promise]] containing `true`, if the destination step can be transitioned to and false otherwise
    */
-  abstract canGoToStep(destinationIndex: number): Promise<boolean>;
+  canGoToStep(destinationIndex: number): Promise<boolean>;
 
   /**
    * Tries to transition to the wizard step, as denoted by the given destination index.
-   * If this is not possible, the current wizard step should be exited and then reentered with `MovingDirection.Stay`
    *
-   * @param destinationIndex The index of the destination step
+   * When entering the destination step, the following actions are done:
+   * - the old current step is set as completed
+   * - the old current step is set as unselected
+   * - the old current step is exited
+   * - the destination step is set as selected
+   * - the destination step is entered
+   *
+   * When the destination step couldn't be entered, the following actions are done:
+   * - the current step is exited and entered in the direction `MovingDirection.Stay`
+   *
+   * @param destinationIndex The index of the destination wizard step, which should be entered
    * @param preFinalize An event emitter, to be called before the step has been transitioned
    * @param postFinalize An event emitter, to be called after the step has been transitioned
    */
-  abstract goToStep(destinationIndex: number, preFinalize?: EventEmitter<void>, postFinalize?: EventEmitter<void>): void;
+  goToStep(destinationIndex: number, preFinalize?: EventEmitter<void>, postFinalize?: EventEmitter<void>): void;
 
   /**
-   * Checks, whether the wizard step, located at the given index, is can be navigated to
+   * Checks, whether the wizard step, located at the given index,
+   * can be navigated to using the navigation bar.
    *
    * @param destinationIndex The index of the destination step
    * @returns True if the step can be navigated to, false otherwise
    */
-  abstract isNavigable(destinationIndex: number): boolean;
+  isNavigable(destinationIndex: number): boolean;
 
   /**
    * Resets the state of this wizard.
    * A reset transitions the wizard automatically to the first step and sets all steps as incomplete.
    * In addition the whole wizard is set as incomplete
    */
-  abstract reset(): void;
+  reset(): void;
 
   /**
    * Tries to transition the wizard to the previous step from the `currentStep`
+   *
+   * @param preFinalize An event emitter, to be called before the step has been transitioned
+   * @param postFinalize An event emitter, to be called after the step has been transitioned
    */
-  goToPreviousStep(preFinalize?: EventEmitter<void>, postFinalize?: EventEmitter<void>): void {
-    if (this.wizardState.hasPreviousStep()) {
-      this.goToStep(this.wizardState.currentStepIndex - 1, preFinalize, postFinalize);
-    }
-  }
+  goToPreviousStep(preFinalize?: EventEmitter<void>, postFinalize?: EventEmitter<void>);
```

Comment:
To make the interface cleaner, we could exclude `goToPrevious/NextStep` methods from it, but users will then have to call `goToStep(wizardState.destinationIndex ± 1)`, which is inconvenient. Also, existing tests will have to be adapted. It feels bad either way. For now, I would choose having these methods in the interface. @madoar, your opinion?

Reply:
No, I can't think of a better idea now. Let's leave the `goToPrevious/NextStep` methods intact.